src/nexus-prisma/NexusPrismaBuilder.ts

```diff
@@ -64,32 +60,23 @@ export class NexusPrismaBuilder {
     }
   }
 
-  getNexusPrismaMethod() {
-    return [
-      this.getCRUDDynamicOutputMethod(),
-      this.getModelDynamicOutputMethod(),
-      ...this.getPrismaScalars(),
-    ]
+  build() {
+    return [this.buildCRUD(), this.buildModel(), ...this.buildScalers()]
   }
 
   /**
    * Generate `t.crud` output method
    */
-  protected getCRUDDynamicOutputMethod() {
+  protected buildCRUD(): DynamicOutputPropertyDef<'crud'> {
     //const methodName = this.params.methodName ? this.params.methodName : 'crud';
-
     return dynamicOutputProperty({
       name: 'crud',
       typeDefinition: `: NexusPrisma<TypeName, 'crud'>`,
       factory: ({ typeDef: t, typeName: graphQLTypeName }) => {
-        if (graphQLTypeName !== 'Query' && graphQLTypeName !== 'Mutation') {
-          throw new Error(
-            `t.crud can only be used on a 'Query' & 'Mutation' objectType. Please use 't.model' instead`,
-          )
-        }
```

Comment:
guard is now a bit lower in an else clause. I like guards, but the pattern seems a bit spontaneous here and felt it became simpler to use `else` after the short-circuit returns were removed.

Reply:
Given that we're in the crud block, and that we know for a fact that it'll only be available in the `Query` and `Mutation` type, I generally like to make that kind of guard high so that it's very clear, and also to make sure that we won't have to care about it anytime later in the code. I guess that's about taste, I'm fine with that too!

Reply:
Yeah I hear you. I'm kind of imagining this could have been done using pattern matching in something like Rust or Haskell, and each branch dispatches except for the `otherwise` where the runtime error is raised. In fact that's close to what you had before!